File: codechef/ABX01.py



t = int(input())
for i in range(t):
    a,n = map(int, input().split())
    #print(a,n)
    num = int(pow(a,n))
    #print("Num is {0}".format(num))
    sum_ = 0

    while(num):
        digit = num % 10
        tmp = 1 + (digit + sum_ -1)% 9
        sum_ = tmp
        num = num // 10
        #print(sum_)
    print(sum_)

File: codechef/ABX01.cpp
#include <bits/stdc++.h>

using namespace std;

int main(void)
{
	long long t;
	cin>>t;
	while(t--)
	{
		uint64_t a,n;
		cin>>a>>n;
		uint64_t sum = 0;
		uint64_t num = pow(a,n);
		//cout<<num<<endl;	
		while(1)
		{
			sum = 0;
			while(num)
			{
				sum += num%10;
				num = num/10;
			}
			if (sum < 10)
				break;
			else
				num = sum;

		}
		cout<<sum<<endl;
	}
}
